src/object/schedcontext.c

```diff
@@ -110,6 +112,18 @@ static exception_t decodeSchedContext_Bind(sched_context_t *sc, extra_caps_t ext
             return EXCEPTION_SYSCALL_ERROR;
         }
 
+        switch (thread_state_get_tsType(TCB_PTR(cap_thread_cap_get_capTCBPtr(cap))->tcbState)) {
```

Comment:
This makes sense. If you are binding to a thread that is blocked to give it more time then it needs enough time to get out of the kernel.